### DISCUSSION TITLE
about Custom domains and etc

### DISCUSSION THREAD

**Discussion by QuirkySiiiiiid** [community/community#135099] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

so i recently get to know that i can upload or deploy my website on github pages and like i see custom domain option... so there is like a thing called dns text record. i want to know how do i set dns txt record? and one more thing the domain provided by the github is it already secured by github?

**Comment by edwinm** [comment]
Yes, you have to set the A record(s) to the ip addresses of GitHub.

Follow the intructions in this link to have your GitHub pages under your domain:

[Configuring a custom domain for your GitHub Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site)

**Reply by QuirkySiiiiiid** [reply]
if i want to yo use the domain github already assigned or the default domain do i have to set dns in that domain also??

**Reply by edwinm** [reply, marked as answer]
You can use the custom domain feature with an domain you have bought before. On the website were you bought the domain, you can set the DNS settings.

Do you already have your own domain?

If not, you can take a look at any domain registrar to see how it works and what the prices are, for example

[namecheap](https://www.namecheap.com/)

**Reply by LiteBrite82** [reply]
@QuirkySiiiiiid 

You dont have to do anything else if you are using the default GitHub domain and are happy with it.
 
If you wanted to get a custom domain name like `<YourCustomName>.com`, you would have to purchase one from a DNS provider. After that, you would connect it to your GitHub Pages site through settings-> Custom Domain, and then configure your DNS settings through your provider like @edwinm posted above.

**Comment by NooruddinLakhani** [comment]
You can purchase a custom domain from https://www.nominus.com for your website. After buying the domain, you will need to configure DNS settings to connect the domain to GitHub Pages, which involves adding DNS records.

**Comment by rahulshiyani-wlu** [comment]
Subject: Unable to Remove Custom Domain rahulshiyani.me from Repository

Description:

Hello GitHub Support Team,

I am experiencing an issue with the custom domain rahulshiyani.me on one of my repositories. Despite my efforts to clear the domain, I am receiving the following error message:

Problem: The custom domain rahulshiyani.me is already taken by another repository in your account.

Here are the actions I have already taken:

Verified that no CNAME file exists in the root directory of my repository.
Cleared the "Custom Domain" field in the Settings > Pages section of all related repositories.
Checked my DNS settings to ensure no records are pointing to GitHub Pages.
Attempted to disassociate the domain from other repositories in my account.
Unfortunately, the issue persists, and I cannot use the default GitHub Pages URL for the repository.

I would appreciate your assistance in:

Removing the custom domain rahulshiyani.me from any repository where it is currently associated in my account.
Reverting my repository to use the default GitHub Pages URL:
https://<username>.github.io/<repository-name>/

Thank you for your time and support. Please let me know if you need additional information or screenshots to assist in resolving this issue.

Best regards,
Rahul

**Reply by wyzh98** [reply]
Same issue. My website is being DNS hijacked and I am trying to recover it. I have cleaned the caches but still failed.
